templates/app-extra.yaml

```diff
@@ -0,0 +1,68 @@
+# This creates and configures extra ra things after deploying beanstalk apps
+# The template should:
+#   - add a certificate to the existing listener
+#   - add a listener rule on port 80 that redirects to correct host on port 443
+#   - add a listener rule on port 443 that redirects to correct host on port 443 if hit as friendlyEndpoint
+AWSTemplateFormatVersion: 2010-09-09
+Description: Beanstalk ELB resources and configs
+Parameters:
+  Listener80Arn:
+    Description: ARN of the ALB http listener created by Beanstalk
+    Type: String
+  Listener443Arn:
+    Description: ARN of the ALB https listerner created by Beanstalk
+    Type: String
+  HttpRedirectTarget:
+    Description: Host name to redirect to
+    Type: String
+  FriendlyEndpoint:
+    Description: Friendly endpoint
+    Type: String
+Resources:
+  # Cloudformation does not support adding multiple certificats in Beanstalk
+  # with aws:elbv2:listener:443 -> SSLCertificateArns therefore we configure
+  # the Beanstalk ELB with a CFN resource in a separate step
+  SSLCertificate:
+    Type: AWS::ElasticLoadBalancingV2::ListenerCertificate
+    Properties:
+      ListenerArn: !Ref Listener443Arn
+      Certificates:
+        - CertificateArn: !ImportValue
+            'Fn::Sub': '${AWS::Region}-synapselogin-shared-SSLCertificateSageIT'
+  HttpRedirectRule:
+    Type: AWS::ElasticLoadBalancingV2::ListenerRule
+    Properties:
+      Actions:
+      - Type: redirect
+        RedirectConfig:
+          Protocol: HTTPS
+          Host: !Ref HttpRedirectTarget
+          Query: '#{query}'
+          Path: '/#{path}'
+          Port: '443'
+          StatusCode: HTTP_302
+      Conditions:
+      - Field: host-header
+        Values:
+          - !Ref HttpRedirectTarget
+          - !Ref FriendlyEndpoint
+      ListenerArn: !Ref Listener80Arn
+      Priority: 1
+  HostRedirectRule:
+    Type: AWS::ElasticLoadBalancingV2::ListenerRule
+    Properties:
+      Actions:
+      - Type: redirect
+        RedirectConfig:
+          Protocol: HTTPS
+          Host: !Ref HttpRedirectTarget
+          Query: '#{query}'
+          Path: '/#{path}'
+          Port: '443'
+          StatusCode: HTTP_302
+      Conditions:
+      - Field: host-header
+        Values:
+        - !Ref FriendlyEndpoint
+      ListenerArn: !Ref Listener443Arn
+      Priority: 1
```

Comment:
should this priority be different than HttpRedirectRule?

Reply:
I think it's fine. They are on different listeners.